annotationProcessor/src/main/java/org/corfudb/annotations/ObjectAnnotationProcessor.java

```diff
@@ -383,8 +383,10 @@ public void generateProxy(TypeElement classElement)
                                         &&
                                 methodSet.stream().noneMatch(x -> x.method.toString()
                                 .equals(method.toString()))) {
-                            methodSet.add(new SmrMethodInfo(method,
-                                            ifaceElement, true));
+                            if (method.getAnnotation(DontInstrument.class) == null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)